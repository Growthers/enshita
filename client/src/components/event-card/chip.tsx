import { memo } from "react";
import styles from "~/styles/components/event-card/chip.module.scss"
import type { ChipProperties } from "./type/model";

// eslint-disable-next-line react/display-name,react/prop-types
const Chip = memo<ChipProperties>(({ status, ...rest}) => {
  let color;
  let content;
  switch(status) {
    case "preparing":
      color = styles["enshita-event-card-status-gray"]
      content = "参加登録受付前"
      break
    case "open":
      color = styles["enshita-event-card-status-green"]
      content = "参加登録受付中"
      break
    case "close":
      color = styles["enshita-event-card-status-gray"]
      content = "参加登録締切済み"
      break
    case "suddenOpen":
      color = styles["enshita-event-card-status-green"]
      content = "飛び入り参加登録受付中"
      break
    case "suddenClose":
      color = styles["enshita-event-card-status-gray"]
      content = "飛び入り参加登録受付終了"
      break
    case "finish":
      color = styles["enshita-event-card-status-gray"]
      content = "イベント終了"
      break
    // no default
  }
  return (
    <div className={`${styles["enshita-event-card-chip"]} ${color}`} {...rest}>
      <p className={styles["enshita-event-card-chip-content"]}>{content}</p>
    </div>
  )
})

export { Chip }
