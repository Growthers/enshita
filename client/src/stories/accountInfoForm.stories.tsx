// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
