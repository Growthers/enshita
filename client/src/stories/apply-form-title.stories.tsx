import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
