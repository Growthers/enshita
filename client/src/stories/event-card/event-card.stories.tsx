import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import EventCard from "~/components/event-card";

type T = typeof EventCard
type Story = ComponentStoryObj<T>

export default {
  args: {
    eventId: "test",
    title: "This is test",
    startDate: "2022-08-27T19300009",
    ogp: "https://ogp.re-taro.dev/api/ogp?title=This%20is%20test",
    status: "open",
    hashTag: "test",
    speakerQuotaTypeList: [{"id":"02GA1BSFZQVZJS5PBADGF8X98E","name":"3分枠","time":3,"currentCount":2,"total":5},{"id":"03GA1BSFZQVZJS5PBADGF8X98E","name":"5分枠","time":5,"currentCount":3,"total":5}]
  }, component: EventCard
} as ComponentMeta<T>

export const Default: Story = {}
