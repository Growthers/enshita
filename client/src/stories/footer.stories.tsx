// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
