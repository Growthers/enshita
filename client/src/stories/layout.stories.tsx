import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
