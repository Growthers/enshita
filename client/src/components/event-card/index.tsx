import type { FC } from "react";
import { Link } from "wouter";
import { useMemo } from "react";
import dayjs from "dayjs";
import styles from "~/styles/components/event-card/event-card.module.scss";
import type { EventCardProperties } from "./type/model";
import { Chip } from "./chip";

const EventCard: FC<EventCardProperties> = ({
  eventId,
  title,
  startDate,
  ogp,
  status,
  hashTag,
  speakerQuotaTypeList,
}) => {
  const memo = useMemo(() => {
    let current = 0;
    let total = 0;
    speakerQuotaTypeList.forEach(speakerQuotaType => {
      current += speakerQuotaType.currentCount;
      total += speakerQuotaType.total;
    });
    return {
      current,
      total,
    };
  }, [speakerQuotaTypeList]);
  return (
    <Link to={`/event/${eventId}`}>
      <div className={styles["enshita-event-card"]}>
        <div className={styles["enshita-event-card-image-wrapper"]}>
          <img
            src={ogp}
            alt={title}
            className={styles["enshita-event-card-image"]}
          />
        </div>
        <div className={styles["enshita-event-card-content-wrapper"]}>
          <div className={styles["enshita-event-card-content"]}>
            <h1 className={styles["enshita-event-card-title"]}>{title}</h1>
            <div className={styles["enshita-event-card-sub-content"]}>
              <time className={styles["enshita-event-card-date"]}>
                {dayjs(startDate).format("MMM DD")}
              </time>
              <p className={styles["enshita-event-card-hash-tag"]}>{hashTag}</p>
            </div>
          </div>
          <div className={styles["enshita-event-card-option"]}>
            <p className={styles["enshita-event-card-num-of-speaker"]}>
              <span className={styles["enshita-event-card-slash"]}>
                {memo.current}
              </span>
              {memo.total}
            </p>
            <Chip status={status} />
          </div>
        </div>
      </div>
    </Link>
  );
};

export default EventCard;
