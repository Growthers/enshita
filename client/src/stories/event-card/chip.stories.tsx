import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import { Chip } from "~/components/event-card/chip";

type T = typeof Chip
type Story = ComponentStoryObj<T>

export default {
  args: {
    status: "preparing"
  }, component: Chip
} as ComponentMeta<T>

export const Defalut: Story = {}
export const Open: Story = {
  args: {
    status: "open"
  }
}
export const Close: Story = {
  args: {
    status: "close"
  }
}
export const SuddenOpen: Story = {
  args: {
    status: "suddenOpen"
  }
}
export const SuddenClose: Story = {
  args: {
    status: "suddenClose"
  }
}
export const Finish: Story = {
  args: {
    status: "finish"
  }
}
