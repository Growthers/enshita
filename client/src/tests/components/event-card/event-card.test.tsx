import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/event-card/event-card.stories";

const { Default } = composeStories(stories);

describe("(components) event card", () => {
  test("take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
