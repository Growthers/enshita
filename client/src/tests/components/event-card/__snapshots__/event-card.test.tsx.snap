// Vitest Snapshot v1

exports[`(components) event card > take snapshot 1`] = `
<div>
  <div
    class="_enshita-event-card_a8aaa5"
    href="/event/test"
  >
    <div
      class="_enshita-event-card-image-wrapper_a8aaa5"
    >
      <img
        alt="This is test"
        class="_enshita-event-card-image_a8aaa5"
        src="https://ogp.re-taro.dev/api/ogp?title=This%20is%20test"
      />
    </div>
    <div
      class="_enshita-event-card-content-wrapper_a8aaa5"
    >
      <div
        class="_enshita-event-card-content_a8aaa5"
      >
        <h1
          class="_enshita-event-card-title_a8aaa5"
        >
          This is test
        </h1>
        <div
          class="_enshita-event-card-sub-content_a8aaa5"
        >
          <time
            class="_enshita-event-card-date_a8aaa5"
          >
            Aug 27
          </time>
          <p
            class="_enshita-event-card-hash-tag_a8aaa5"
          >
            test
          </p>
        </div>
      </div>
      <div
        class="_enshita-event-card-option_a8aaa5"
      >
        <p
          class="_enshita-event-card-num-of-speaker_a8aaa5"
        >
          <span
            class="_enshita-event-card-slash_a8aaa5"
          >
            5
          </span>
          10
        </p>
        <div
          class="_enshita-event-card-chip_f96697 _enshita-event-card-status-green_f96697"
        >
          <p
            class="_enshita-event-card-chip-content_f96697"
          >
            参加登録受付中
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
